Feature/post graphql integration revisions

rev-development opened the pull request.

rev-development committed:
Fix: Form Elements throw Hydration Mismatch errors w/o id

This seems to be a bug in vuetify/nuxt, for some reason it has trouble keying the elements

See: https://github.com/vuetifyjs/vuetify/issues/19015

rev-development committed:
Fix: Equipment Data is fetched after Login

So the problem was that after logging in, the equipment data would not be fetched. However it was being fetched on page refresh.

After a lot of tinkering I figured out that:
1. Data that is present on first load needs to be fetched with useAsyncQuery()

2. This needs to be done inside a page/component/composable. Anything wrapping or using useAsyncData() needs to be sitting inside something actually touching nuxt. I figured this out because copy-pasting the original non-async code threw lifecycle errors.

rev-development committed:
Added middleware for auth

rev-development committed:
Swapped position of attributes list and OEE subcomponents

rev-development committed:
Added composable for equip detail page, still need to swap code

scott181182 commented on frontend/composables/useAsyncEquipmentIds.ts:
```diff
@@ -0,0 +1,53 @@
+    "default",
+    { errorPolicy: "ignore" }
+    , {
+      transform: (typeRes) => {
```
Because of the complexity of this function, let's extract it to a separate function in this file that we can pass to this object field.

rev-development replied:
Removing the computeds has simplified it significantly. I think now, unless we're using it elsewhere, being in a composable is as broken out as it needs to be.

scott181182 replied:
I still think putting this code in a separate function (in this file) would aid readability, but consider it non-blocking for this PR

scott181182 commented on frontend/composables/useAsyncEquipmentIds.ts:
```diff
@@ -0,0 +1,53 @@
+    , {
+      transform: (typeRes) => {
+      /** All equipment with OEE `equipmentType`. */
+        const oeeEquipment = computed(() => unique(
```
Now that you're using the `transform` function, none of this functionality should use reactive component. We should remove all of these `computed` calls

scott181182 commented on frontend/components/metric-progress-circular.vue:
```diff
@@ -35,5 +35,8 @@ const { label, value, size, width } = withDefaults(defineProps<Props>(), {
-const progressValue = computed(() => typeof value === "number" ? clamp(value, { min: 0, max: 100 }) : 0);
+const progressValue = computed(() => typeof value === "number"
+  ? clamp(value, { min: 0,
+    max: 100 })
```
This, along with a few other things in this PR, seem to have weird formatting. I think most of your linting changes are good additions, but allowing this newline inconsistency seems _wrong_.

scott181182 commented on frontend/pages/equipment/[equipmentID].vue:
```diff
@@ -19,11 +19,35 @@
 
-      <v-col cols="auto" class="pr-0">
-        <v-tabs v-model="activeTabLabel" direction="vertical" class="border-md border-e-0">
+      <v-col>
```
Should the column width be specified here?

scott181182 commented on frontend/pages/equipment/[equipmentID].vue:
```diff
@@ -142,7 +121,7 @@ const metricTabs = computed(() => [
   {
     label: "Quality",
-    color: "yellow",
+    color: "indigo",
```
:O

rev-development replied:
The yellow was difficult to view on the light background, and going darker wasn't an option because of machine state colors

scott181182 commented on frontend/pages/index.vue:
```diff
@@ -2,37 +2,25 @@
-          :equipment="equipment"
-          @click="navigateTo(`/equipment/${equipment.id}`)"
-        />
+        <template v-if="equipments">
```
Could we add a `v-else` to show the user if there aren't any equipment? Or if any are loading?

scott181182 commented on frontend/pages/login.vue:
```diff
@@ -55,6 +55,7 @@ async function onSubmit() {
           <v-card :loading="loading">
             <v-card-text>
               <v-text-field
+                id="authenticator"
```
Can we also add `name` attributes to these fields?

scott181182 reviewed (changes requested):
Excellent space saving and code splitting by utilizing composables and middleware. I haven't looked at it on my machine yet, but left a few comments on code quality

scott181182 committed:
style(frontend): added lint to make object style more consistent

scott181182 committed:
chore(frontend): removed redundant eslint import plugin package, and downgraded eslint to work with other package managers

scott181182 committed:
feat(frontend): changed column widths

scott181182 committed:
feat(frontend): changed initial time series request to work better with expressions

scott181182 committed:
feat(frontend): added ability to customize label class on circular metrics

rev-development committed:
Added some linting rules, still not quite what I want but no more jumbo chain lines

rev-development committed:
Update useAsyncEquipmentIds.ts

rev-development committed:
Added name properties to login form fields

rev-development committed:
Update login.vue

rev-development committed:
Added v-else for when no data is present

rev-development committed:
Converted equipment detail page to use async

The problem with the OEE Dial appears to be related to the data availability on page load, this makes it better but there are still strange issues occuring.
What I thought was originally the data not being present on page load was actually vue-router stalling and no route being selected.

rev-development committed:
Reactivity on columns

scott181182 committed:
chore(frontend): added function names for composables

scott181182 committed:
refactor(frontend): exposed root refresh and status for equipment with OEE

scott181182 committed:
feat(frontend): added more status feedback on dashboard

scott181182 committed:
refactor(frontend): improved responsiveness on dashboard

scott181182 committed:
refactor(frontend): updated time series query variables to better support expressions

scott181182 committed:
feat(frontend): improved responsiveness on equipment detail page and fixed callbacks

scott181182 committed:
feat(frontend): added function for rendering datetimes

scott181182 committed:
refactor(frontend): another attempt at fixing empty data

scott181182 committed:
refactor(frontend): extracted attribute values and tables to separate components

scott181182 committed:
fix(frontend): fixed some bad classes

rev-development committed:
Added script

scott181182 commented:
Seemingly fixes #19

scott181182 committed:
chore(scripts): added start script for built server

scott181182 committed:
revert(frontend): removed refresh buttons since caching seems to prevent them from working

We can re-add in the future

scott181182 committed:
Merge branch 'feature/post-graphql-integration-revisions' of github.com:cesmii/OEE-SMIP-App into feature/post-graphql-integration-revisions

scott181182 committed:
chore: added/updated some package metadata

scott181182 committed:
feat(frontend): added better 'no value' render to equipment card
